generate-manifest

```diff
@@ -0,0 +1,401 @@
+
+logging.basicConfig(format="%(levelname)s: %(message)s")
+logger = logging.getLogger(__name__)
+
+KOLIBRI_CONTENT_EXTENSION_BASE_ID = "org.learningequality.Kolibri.Content"
+MANIFEST = None
+
+BASE_PATH = os.path.realpath(os.path.dirname(__file__))
+MANIFEST_TEMPLATE_PATH = os.path.join(BASE_PATH, "manifest.json.in")
+METAINFO_PATH = os.path.join(BASE_PATH, "metainfo.xml")
+EXTENSIONS_CATALOG_PATH = os.path.join(BASE_PATH, "extensions-catalog.json")
+KOLIBRI_HOME_PATH = os.path.join(BASE_PATH, "kolibri_home")
+CONTENT_DIR_PATH = os.path.join(KOLIBRI_HOME_PATH, "content")
+CONTENT_JSON_PATH = os.path.join(CONTENT_DIR_PATH, "content.json")
+
+METAINFO_DEFAULTS = {
+    "metadata_license": "CC0-1.0",
+    "url_homepage": "https://learningequality.org/",
+    "project_group": "Learning Equality",
+    "developer_name": "Learning Equality",
+    "categories": ["Education"],
+}
+
+
+def run_command(args, **kwargs):
+    logger.info("Running command: {}".format(args))
+    subprocess.check_call(args, **kwargs)
+
+
+def install_kolibri_flatpak():
+    # This has to be done like this to get the same Kolibri
+    # as org.learningequality.Kolibri does, so we avoid repeating
+    # the Kolibri installation procedure.
+    run_command(
+        [
+            "flatpak",
+            "remote-add",
+            "--if-not-exists",
+            "flathub",
+            "https://flathub.org/repo/flathub.flatpakrepo",
+        ]
+    )
+    run_command(
+        [
+            "flatpak",
+            "install",
+            "--noninteractive",
+            "-y",
+            "--or-update",
+            "flathub",
+            get_kolibri_flatpak_ref(),
+        ]
+    )
+
+
+def run_kolibri(args):
+    run_command(
+        [
+            "flatpak",
+            "run",
+            "--command=/app/libexec/kolibri",
+            "--filesystem={}".format(BASE_PATH),
+            get_kolibri_flatpak_ref(),
+        ]
+        + args,
+        env={"KOLIBRI_HOME": KOLIBRI_HOME_PATH},
+    )
+
+
+def format_kolibri_flags(channel, flags_to_use):
+    flags = []
+
+    for flag_name in flags_to_use:
+        flag = "--{}".format(flag_name)
+        flag_value = channel.get(flag_name)
+        if type(flag_value) == bool:
+            flags.append(flag)
+        elif type(flag_value) == str:
+            flags.extend([flag, flag_value])
+        elif type(flag_value) == list:
+            flags.extend([flag, ",".join(flag_value)])
+
+    return flags
+
+
+def import_channel(channel):
+    origin = channel.get("origin", "network")
+    origin_args = None
+    if origin == "network":
+        origin_args = ["network", channel["channel_id"]]
+    elif origin == "disk":
+        origin_args = ["network", channel["channel_id"], channel["directory"]]
+
+    run_kolibri(
+        ["manage", "importchannel"]
+        + format_kolibri_flags(channel, ["baseurl", "no_upgrade"])
+        + origin_args
+    )
+
+    if len(channel.get("node_ids", [])) > 0:
```

Comment:
I don't think this is right. I shouldn't be required to list all node IDs in the case where I want to import all content for a channel.

I think this is why when I tried to create an extension, it ended up not having any of the content from that channel in it.

```
$ ./generate-manifest --id Sinugun_samod_hujos --channel channel_id=d69e725e8aea4d19ad58d41c2b4dc315
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Invoking command importchannel network d69e725e8aea4d19ad58d41c2b4dc315
INFO     Downloading data for channel id d69e725e8aea4d19ad58d41c2b4dc315
INFO     Importing ContentTag data
INFO     Importing ContentNode_has_prerequisite data
INFO     Importing ContentNode_related data
INFO     Importing ContentNode_tags data
INFO     Importing ContentNode data
INFO     Importing Language data
INFO     Importing File data
INFO     Importing LocalFile data
INFO     Importing AssessmentMetaData data
INFO     Importing ChannelMetadata data
INFO     Channel d69e725e8aea4d19ad58d41c2b4dc315 successfully imported into the database

wjt@camille:~/src/flathub/org.learningequality.Kolibri$ flatpak run --command=bash app/org.learningequality.Kolibri/x86_64/master
[📦 org.learningequality.Kolibri ~]$ find /app/share/kolibri-content/
/app/share/kolibri-content/
/app/share/kolibri-content/Sinugun_samod_hujos
/app/share/kolibri-content/Sinugun_samod_hujos/.ref
/app/share/kolibri-content/Sinugun_samod_hujos/manifest.json
/app/share/kolibri-content/Sinugun_samod_hujos/content
/app/share/kolibri-content/Sinugun_samod_hujos/content/databases
/app/share/kolibri-content/Sinugun_samod_hujos/content/databases/d69e725e8aea4d19ad58d41c2b4dc315.sqlite3
/app/share/kolibri-content/Sinugun_samod_hujos/content/content.json
/app/share/kolibri-content/Sinugun_samod_hujos/share
/app/share/kolibri-content/Sinugun_samod_hujos/share/metainfo
/app/share/kolibri-content/Sinugun_samod_hujos/share/app-info
/app/share/kolibri-content/Sinugun_samod_hujos/share/app-info/xmls
/app/share/kolibri-content/Sinugun_samod_hujos/share/app-info/xmls/org.learningequality.Kolibri.Content.Sinugun_samod_hujos.xml.gz
/app/share/kolibri-content/Sinugun_samod_hujos/share/appdata
/app/share/kolibri-content/Sinugun_samod_hujos/share/appdata/org.learningequality.Kolibri.Content.Sinugun_samod_hujos.appdata.xml
```

And although there are messages about it from Kolibri in the journal, no channels appear in the interface.

Reply:
Yep, if I manually specify a node ID it works.

Reply:
I'm not sure why I had the memory of thinking that not specifying `node_ids` wouldn't download anything, I tried it and it works, gonna fix it.

Reply:
Ok, I'm not crazy, content is not being imported when not using `node_ids` AND after deleting `kolibri_home/content`, looks like you also have to run `deletechannel` and `deletecontent` to really mark the content as deleted. I think I'm gonna do it with `exportchannel` and `exportcontent`, it's the ideal and it's not hard to do.